### ISSUE TITLE
https://hookname.hook.io invalid SSL certificate

### ISSUE THREAD

**Issue by zanona** [bigcompany/hook.io#219]
Apparently hook.io subdomains have invalid SSL certificate attached to it? Apparently the one being used is not a wildcard?

Please see attached 
![screen shot 2016-04-15 at 13 35 04](https://cloud.githubusercontent.com/assets/32017/14561589/2c29a650-030f-11e6-88fe-0f2ef78b72bb.png)

**Comment by Marak** [comment]
Yeah, we need to buy a wildcard SSL cert.

I guess I could just buy one now? I should do that.

**Comment by Marak** [comment]
Tracking:

#102 
#101

**Comment by Marak** [comment]
Duplicate #101

**Comment by zanona** [comment]
Thanks @Marak. I think it will be a good addition to hook.io. :+1:

**Comment by Marak** [comment]
I'll keep it open until I buy and deploy SSL cert.

Any suggestions from where to buy unlimited wildcard cert?

**Comment by zanona** [comment]
@Marak [This may give you some good options](https://www.namecheap.com/security/ssl-certificates/wildcard.aspx) :) Comodo seems a nice one

**Comment by zanona** [comment]
Hey @Marak, I was doing some research and found about this initiative. It might be worth checking it out.
It could be a good idea to not only use it for the wildcard one but also to provide SSL for all custom domains without any additional hassle. https://letsencrypt.org

Currently, Netlify works with Let's Encrypt to provide SSL for custom domains. It seems to be great and much better than cloud flare's SNI where you have a list of other clients listed in the certificate description.

https://www.netlify.com/blog/2016/01/15/free-ssl-on-custom-domains

**Comment by Marak** [comment]
@zanona -

I wouldn't mind not paying for SSL certs. The wildcard from comodo is over $100

**Comment by zanona** [comment]
Yes, I believe this really shines over custom domains. It would add so much value to the service, since it makes a lot of sense for APIs to only accept HTTPS requests. Imagine if that can be offered with every hook. It would be crazy cool.
Apparently the way it works, is by running their CLI generating the certs and validating against their server every 3 months — You could even create a hook to achieve that :).

It's seems to be very simple as long as there's control over the server it's running on. In which case, you may have :)

**Comment by Marak** [comment]
see: 

https://github.com/Daplie/node-letsencrypt
https://github.com/Daplie/letsencrypt-express

**Comment by zanona** [comment]
Awesomeness!!!
![Awesome](http://i.giphy.com/bE5qi2tiSfdZu.gif)

**Comment by reimertz** [comment]
any news on this?

**Comment by ibrado** [comment]
The good news, is next month (Jan 2018), LetsEncrypt will be [releasing wilcard certificates](https://letsencrypt.org/2017/07/06/wildcard-certificates-coming-jan-2018.html)! This could finally be closed. :smile:

**Comment by Marak** [comment]
That's really good news about letsencrypt issuing wildcard @ibrado 

We recently switched to LE. I'll get a wildcard cert as soon as they will issue it.

Should help in getting the subdomains secured.
